Sources/CNIOLinux/shim.c

```diff
@@ -188,4 +188,17 @@ int CNIOLinux_system_info(struct utsname* uname_data) {
 
 const unsigned long CNIOLinux_IOCTL_VM_SOCKETS_GET_LOCAL_CID = IOCTL_VM_SOCKETS_GET_LOCAL_CID;
 
+const char* CNIOLinux_dirent_dname(struct dirent* ent) {
+  return ent->d_name;
+}
+
+int CNIOLinux_renameat2(int oldfd, const char* old, int newfd, const char* newName, unsigned int flags) {
+  return renameat2(oldfd, old, newfd, newName, flags);
```

Comment:
This [broke my Android CI today](https://github.com/finagolfin/swift-android-sdk/actions/runs/7553358697/job/20564040487#step:13:71) because Android didn't add this to [its Bionic libc till API 30 a couple years ago](https://android.googlesource.com/platform/bionic/+/9c06d16ca3d230059ccf3b7f270574bc5e8fcfcb/libc/include/stdio.h#199), whereas I build against the older API 24.

Since this approach is still experimental, can we just disable `rename()` alone on Android or something like that?

Reply:
This is also used during file creation so we can't just disable it on Android. I think we can work around this though as this also isn't available on Darwin so I think we can just do what we're doing on Darwin.

Reply:
See https://github.com/apple/swift-nio/pull/2627